### ISSUE TITLE
✅ 🐘 🐞 Using ${var} in strings is deprecated, use {$var} instead

### ISSUE THREAD

**Issue by peixotorms** [elementor/elementor#24246]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When I run a wp-cli command, I get this error:

```
PHP: 2023-11-11 20:37:47 [notice X 0][/home/sites/***/www/wp-content/plugins/elementor-pro/core/editor/editor.php::129] Using ${var} in strings is deprecated, use {$var} instead [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
```

### Steps to reproduce

run `wp term recount`

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt




== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: Percona Server (GPL), Release '26', Revision '0fe62c85' v8.0.34-26
	PHP Version: 8.2.11
	PHP Memory Limit: 1024M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.4.1
	Site URL: https://***
	Home URL: https://***
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 1024M
	Max Memory limit: 1024M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 2.0.0
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.9.0
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36

== Active Plugins ==
	ACF Customizations and Shortcodes
		Version: 1.0.0
		Author: Raul Peixoto

	Advanced Custom Fields PRO
		Version: 6.2.2
		Author: WP Engine

	bbPress
		Version: 2.6.9
		Author: The bbPress Contributors

	Disable Comments
		Version: 2.4.5
		Author: WPDeveloper

	Disable Gutenberg
		Version: 3.1
		Author: Jeff Starr

	Duplicate Page
		Version: 4.5.3
		Author: mndpsingh287

	DynamicTags
		Version: 1.2.2
		Author: RTO GmbH

	Elementor
		Version: 3.17.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.17.1
		Author: Elementor.com

	Essential Addons for Elementor
		Version: 5.8.16
		Author: WPDeveloper

	Essential Addons for Elementor - Pro
		Version: 5.8.2
		Author: WPDeveloper

	FacetWP
		Version: 4.2.7
		Author: FacetWP, LLC

	FacetWP - Elementor
		Version: 1.8.1
		Author: FacetWP, LLC

	Formidable ACF
		Version: 1.0.1
		Author: Strategy11

	Formidable Forms
		Version: 6.5.4
		Author: Strategy11 Form Builder Team

	Formidable Forms Pro
		Version: 6.5.4
		Author: Strategy11

	Formidable Geolocation
		Version: 1.1.1
		Author: Strategy11

	Simple Custom CSS and JS
		Version: 3.45
		Author: SilkyPress.com

	Site Reviews
		Version: 6.11.4
		Author: Paul Ryley

	Templately
		Version: 2.2.9
		Author: Templately

	Ultimate Member
		Version: 2.7.0
		Author: Ultimate Member

	Ultimate Member - bbPress
		Version: 2.1.5
		Author: Ultimate Member

	Ultimate Member - Followers
		Version: 2.2.9
		Author: Ultimate Member

	Ultimate Member - Private Messages
		Version: 2.3.6
		Author: Ultimate Member

	Ultimate Member - Profile Completeness
		Version: 2.2.7
		Author: Ultimate Member

	Ultimate Member - Profile tabs
		Version: 1.0.9
		Author: Ultimate Member

	Ultimate Member - reCAPTCHA
		Version: 2.3.4
		Author: Ultimate Member

	Ultimate Member - User Reviews
		Version: 2.2.4
		Author: Ultimate Member

	Ultimate Member - Verified Users
		Version: 2.1.9
		Author: Ultimate Member

	WP All Import - ACF Add-On
		Version: 3.3.8
		Author: Soflyy

	WP All Import - User Import Add-On Pro
		Version: 1.1.8
		Author: Soflyy

	WP All Import Pro
		Version: 4.8.5
		Author: Soflyy

	WP Mail Logging
		Version: 1.12.0
		Author: WP Mail Logging Team


== Elements Usage ==
	
	archive : 3
		archive-posts : 1
		author-box : 1
		container : 9
		heading : 4
		image : 2
		shortcode : 3
	footer : 1
		container : 5
		heading : 5
		html : 1
		image-box : 1
		nav-menu : 1
		social-icons : 1
	header : 1
		container : 3
		nav-menu : 1
		theme-site-logo : 1
	wp-page : 12
		button : 10
		container : 81
		eael-content-timeline : 1
		eael-counter : 4
		eael-testimonial : 6
		google_maps : 1
		heading : 57
		icon-box : 5
		icon-list : 10
		image : 9
		image-carousel : 5
		shortcode : 18
		text-editor : 4
		toggle : 2
	single-post : 4
		container : 14
		eael-adv-tabs : 1
		google_maps : 1
		heading : 6
		post-info : 4
		share-buttons : 4
		shortcode : 2
		theme-post-content : 1
		theme-post-title : 3
		toggle : 1
	single-page : 1
		container : 2
		shortcode : 1
	loop-item : 2
		container : 2
		eael-business-reviews : 1
		heading : 2
		site_reviews : 1
		site_reviews_form : 1
		site_reviews_summary : 1


== Settings ==
	
	cpt_support: post, page
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap
	meta_generator_tag: 1


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active by default
	Upgrade Swiper Library: Active by default
	Grid Container: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header &amp; Footer: Active by default
	Editor Top Bar: Inactive by default
	Optimized Gutenberg Loading: Inactive by default
	Build with AI: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive by default
	Optimize Image Loading: Inactive by default
	Global Style Guide: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Inactive by default
	Taxonomy Filter: Inactive by default


== Log ==
	
Log: showing 20 of 212023-10-28 17:19:40 [info] Elementor/Upgrades - _on_each_version Start  
2023-10-28 17:19:40 [info] Elementor/Upgrades - _on_each_version Finished 
2023-10-28 17:19:40 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Start  
2023-10-28 17:19:40 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Finished 
2023-10-28 17:19:40 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.16.6',
  'to' => '3.17.1',
)]
2023-11-01 11:49:10 [info] elementor::elementor_updater Started 
2023-11-01 11:49:10 [info] Elementor/Upgrades - _on_each_version Start  
2023-11-01 11:49:10 [info] Elementor/Upgrades - _on_each_version Finished 
2023-11-01 11:49:10 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Start  
2023-11-01 11:49:10 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Finished 
2023-11-01 11:49:10 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.17.1',
  'to' => '3.17.2',
)]
2023-11-01 12:27:15 [info] elementor-pro::elementor_pro_updater Started 
2023-11-01 12:27:15 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-11-01 12:27:15 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-11-01 12:27:15 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.17.0',
  'to' => '3.17.1',
)]
2023-11-11 17:42:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.17.2',
  'to' => '3.17.3',
)]
2023-11-11 17:42:05 [info] elementor::elementor_updater Started 
2023-11-11 17:42:05 [info] Elementor/Upgrades - _on_each_version Start  
2023-11-11 17:42:06 [info] Elementor/Upgrades - _on_each_version Finished 
2023-11-11 17:42:06 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.17.2',
  'to' => '3.17.3',
)]

PHP: showing 12 of 12PHP: 2023-10-28 17:19:40 [notice X 10253][../wp-content/plugins/elementor-pro/core/editor/editor.php::129] Using ${var} in strings is deprecated, use {$var} instead [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-10-28 17:22:59 [notice X 10397][../wp-content/plugins/elementor/includes/elements/container.php::43] Creation of dynamic property Elementor\Includes\Elements\Container::$logical_dimensions_inline_end is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-10-28 22:01:04 [warning X 2][../wp-content/plugins/elementor-pro/modules/theme-elements/widgets/post-info.php::833] Attempt to read property &quot;name&quot; on array [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-10-29 19:53:40 [warning X 4][../wp-content/plugins/elementor-pro/modules/theme-builder/conditions/post.php::35] Attempt to read property &quot;name&quot; on null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-10-31 22:40:11 [notice X 8][../wp-content/plugins/elementor-pro/modules/screenshots/module.php::33] Function utf8_decode() is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-11-01 15:04:32 [warning X 4][../wp-content/plugins/elementor/core/logger/items/js.php::17] Undefined array key &quot;url&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-11-01 22:30:21 [notice X 9][../wp-content/plugins/elementor/includes/elements/container.php::42] Creation of dynamic property Elementor\Includes\Elements\Container::$logical_dimensions_inline_start is deprecated [array (
  'trace' => '
#0: ../wp-content/plugins/elementor/includes/elements/container.php(42): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ../wp-content/plugins/elementor/includes/managers/elements.php(257): Elementor\Includes\Elements\Container -> __construct()
#2: ../wp-content/plugins/elementor/includes/managers/elements.php(192): Elementor\Elements_Manager -> init_elements()
#3: ../wp-content/plugins/elementor/includes/managers/elements.php(76): Elementor\Elements_Manager -> get_element_types()
#4: ../wp-content/plugins/elementor/includes/template-library/sources/base.php(315): Elementor\Elements_Manager -> create_element_instance()
',
)]
PHP: 2023-11-01 22:30:21 [notice X 9][../wp-content/plugins/elementor/includes/elements/container.php::43] Creation of dynamic property Elementor\Includes\Elements\Container::$logical_dimensions_inline_end is deprecated [array (
  'trace' => '
#0: ../wp-content/plugins/elementor/includes/elements/container.php(43): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ../wp-content/plugins/elementor/includes/managers/elements.php(257): Elementor\Includes\Elements\Container -> __construct()
#2: ../wp-content/plugins/elementor/includes/managers/elements.php(192): Elementor\Elements_Manager -> init_elements()
#3: ../wp-content/plugins/elementor/includes/managers/elements.php(76): Elementor\Elements_Manager -> get_element_types()
#4: ../wp-content/plugins/elementor/includes/template-library/sources/base.php(315): Elementor\Elements_Manager -> create_element_instance()
',
)]
PHP: 2023-11-04 16:46:05 [warning X 2][../wp-content/plugins/elementor/includes/db.php::267] foreach() argument must be of type array|object, null given [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-11-04 16:49:47 [warning X 14][../wp-content/plugins/elementor/includes/elements/container.php::1847] Undefined array key &quot;size&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-11-05 22:42:41 [warning X 14][../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key &quot;condition_type&quot; [array (
  'trace' => '
#0: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: ../wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2023-11-05 22:43:04 [warning X 2][../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key &quot;condition_type&quot; [array (
  'trace' => '
#0: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(83): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ../wp-content/plugins/elementor/data/base/endpoint.php(166): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> update_item()
#3: ../wp-content/plugins/elementor/data/base/endpoint.php(307): Elementor\Data\Base\Endpoint -> base_callback()
#4: ../wp-includes/rest-api/class-wp-rest-server.php(1194): Elementor\Data\Base\Endpoint -> Elementor\Data\Base\{closure}()
',
)]

JS: showing 9 of 9JS: 2023-10-29 11:19:18 [error X 3][***/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.17.1:3:805504] Cannot read properties of undefined (reading &#039;toLowerCase&#039;) 
JS: 2023-10-29 14:18:50 [error X 3][***/wp-admin/load-scripts.php?c=1&amp;loadchunk_0=jquery-core,jquery-migrate,utils&amp;ver=6.3.2:2:28722] Cannot read properties of undefined (reading &#039;value&#039;) 
JS: 2023-11-01 14:05:31 [error X 3324][***/wp-content/plugins/elementor/assets/lib/pickr/pickr.min.js?ver=1.5.0:2:14799] Cannot read properties of null (reading &#039;clone&#039;) 
JS: 2023-11-01 14:05:36 [error X 66][***/wp-content/plugins/elementor/assets/lib/pickr/pickr.min.js?ver=1.5.0:2:19552] Cannot read properties of null (reading &#039;changestop&#039;) 
JS: 2023-11-02 08:29:18 [error X 2][***/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.17.2:3:639997] Cannot read properties of undefined (reading &#039;isDesignable&#039;) 
JS: 2023-11-02 12:59:25 [error X 9][***/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.17.2:3:689408] this.model.isValidChild is not a function 
JS: 2023-11-04 00:43:43 [error X 1][***/wp-content/plugins/elementor/assets/js/editor-modules.min.js?ver=3.17.2:3:14266] An invalid argument supplied as multiple control value [array (
  'category' => 'InvalidElementData',
  'tag' => 'Element `undefined` got as `eael_read_more_text_link` value. Expected array or object.',
)]
JS: 2023-11-04 22:56:06 [error X 12][***/wp-content/plugins/elementor/assets/js/frontend.min.js?ver=3.17.2:2:28107] Cannot read properties of undefined (reading &#039;closest&#039;) 
JS: 2023-11-05 23:20:36 [error X 1][***/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.17.2:3:953240] Cannot read properties of undefined (reading &#039;settings&#039;) 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible
	Essential Addons for Elementor: Compatible
	Essential Addons for Elementor - Pro: Compatible
	FacetWP - Elementor: Compatibility not specified

== Elementor Pro - Compatibility Tag ==
	
	Essential Addons for Elementor: Compatible
	Essential Addons for Elementor - Pro: Compatible


````

</details>

**Comment by doronwolf** [comment]
Elementor v3.18 is supposed to have a fix for this issue

**Comment by knofte** [comment]
@doronwolf When will 3.18 be publically available? Lots of sites now with this issue after updating to php8.2 after 8.0 end of active support.

**Comment by doronwolf** [comment]
Hi @knofte 
3.18 is already available right away as a beta version,
if you prefer waiting for the stable version then, I can't guarantee it's the final release date but the official release is supposed to be on Monday (Dec 4)

**Comment by knofte** [comment]
I prefer stable as we do updates via 
wp plugin update

I'll wait :)

Thank you :+1:

**Comment by nicholaszein** [comment]
This issue was solved with version 3.18.
